tests/files/test-image-sizes.php

```diff
@@ -546,27 +544,31 @@ public function test__custom_size() {
 		$width            = 200;
 		$height           = 180;
 		add_image_size( $custom_size_name, $width, $height, true );
-		$attachment_id = self::factory()->attachment->create_object(
-			$this->test_image, 0, [
-				'post_mime_type' => 'image/jpeg',
-				'post_type'      => 'attachment',
-			]
-		);
-		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
-
-		$metadata = wp_get_attachment_metadata( $attachment_id );
-
-		$this->assertEquals( array_merge( array_keys( $this->default_sizes() ), [ $custom_size_name ] ), array_keys( $metadata['sizes'] ), 'The newly registered image size has not been created.' );
-
-		// Has all the sizes, including the new one, have correct dimensions?
-		foreach ( $this->default_sizes() as $size => $properties ) {
-			$this->assertEquals( $properties['calculated_dimensions']['width'], $metadata['sizes'][ $size ]['width'], 'Incorrect calculated width.' );
-			$this->assertEquals( $properties['calculated_dimensions']['height'], $metadata['sizes'][ $size ]['height'], 'Incorrect calculated height.' );
+		try {
+			$attachment_id = self::factory()->attachment->create_object(
+				$this->test_image, 0, [
+					'post_mime_type' => 'image/jpeg',
+					'post_type'      => 'attachment',
+				]
+			);
+			wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+			$metadata = wp_get_attachment_metadata( $attachment_id );
+
+			$this->assertEquals( array_merge( array_keys( $this->default_sizes() ), [ $custom_size_name ] ), array_keys( $metadata['sizes'] ), 'The newly registered image size has not been created.' );
+
+			// Has all the sizes, including the new one, have correct dimensions?
+			foreach ( $this->default_sizes() as $size => $properties ) {
+				$this->assertEquals( $properties['calculated_dimensions']['width'], $metadata['sizes'][ $size ]['width'], 'Incorrect calculated width.' );
+				$this->assertEquals( $properties['calculated_dimensions']['height'], $metadata['sizes'][ $size ]['height'], 'Incorrect calculated height.' );
+			}
+
+			// Does the custom size have the correct dimensions?
+			$this->assertEquals( $height, $metadata['sizes']['custom_size']['height'], 'Incorrect calculated height for the custom size.' );
+			$this->assertEquals( $width, $metadata['sizes']['custom_size']['width'], 'Incorrect calculated width for the custom size.' );
+		} finally {
+			remove_image_size( $custom_size_name );
 		}
```

Comment:
WordPress does not backup/restore `$_wp_additional_image_sizes`, this is why we have to remove what we have added; if we don't, and the tests run in a random order, they can fail.